From 9ed04183d33d37ffe6eaa0e8f9a8696c0dcd3ddc Mon Sep 17 00:00:00 2001
From: Kurt Kiefer <kekiefer@gmail.com>
Date: Tue, 1 Nov 2022 10:47:26 -0700
Subject: [PATCH] Skip map frame in pad prepare to fix spurious warnings

When running with gstreamer message level WARN, the following
message is repeated continuously:

videoaggregator gstvideoaggregator.c:528:gst_video_aggregator_convert_pad_prepare_frame:<comp> Could not map input buffer

This element doesn't need to have video data buffers mapped in
order to do its job. In fact the video data buffers used internally
don't contain video data at all, but instead the nvbuffer pointers,
which are obtained differently.

Signed-off-by: Kurt Kiefer <kekiefer@gmail.com>
---
 gstnvcompositor.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/gstnvcompositor.c b/gstnvcompositor.c
index 67965b5..26ea1c3 100644
--- a/gstnvcompositor.c
+++ b/gstnvcompositor.c
@@ -526,10 +526,7 @@ gst_nvcompositor_pad_prepare_frame (GstVideoAggregatorPad * pad,
     GstVideoAggregator * vagg, GstBuffer * buffer,
     GstVideoFrame * prepared_frame)
 {
-  return
-      GST_VIDEO_AGGREGATOR_PAD_CLASS
-      (gst_nvcompositor_pad_parent_class)->prepare_frame (pad, vagg, buffer,
-      prepared_frame);
+  return TRUE;
 }
 
 /**
